src/hooks.js

```diff
@@ -29,3 +30,49 @@ export function useInstance(initialValue) {
 
   return ref.current;
 }
+
+export function useSingleton({
+  className,
+  ignoreAttributes = true,
+  onComponent = noop => noop,
+  ...restOfNativeProps
+} = {}) {
+  const component = useInstance({
+    instance: null,
+    instances: [],
+    renders: 1,
+  });
+
+  const props = {
+    ignoreAttributes,
+    ...restOfNativeProps,
+  };
+
+  useIsomorphicLayoutEffect(() => {
+    component.instance = createSingleton(component.instances, props);
+
+    onComponent(component);
+
+    return () => {
+      component.instance.destroy();
+      component.instances = component.instances.filter(
+        i => !i.state.isDestroyed,
+      );
```

Comment:
Small nit: `const instance = ...` so you can use the minified local variable in the cleanup function below. Assign `component.instance = instance` right below it (consistency with `<TippySingleton />`